corehq/apps/userreports/pillow.py

```diff
@@ -49,6 +49,31 @@ def timed(*args, **kw):
     return timed
 
 
+def _filter_by_hash(configs, ucr_division):
+    ucr_start = ucr_division[0]
+    ucr_end = ucr_division[-1]
+    filtered_configs = []
+    for config in configs:
+        table_hash = hashlib.md5(config.table_id).hexdigest()[0]
+        if ucr_start <= table_hash <= ucr_end:
+            filtered_configs.append(config)
+    return filtered_configs
+
+
+def _exclude_missing_domains(configs):
+    from corehq.apps.es import DomainES
+    from corehq.elastic import ESError
+
+    config_domains = {conf.domain for conf in configs}
```

Comment:
yeah we were trying to stuff 10k ids into an ES query. it's been an issue on and off, i think it just increases the time it takes to process the query and sometimes causes it to timeout